WordPress/gutenberg
# Allow overriding the iOS platform destination for RN E2E testing

## What?

The core change in this PR is adding an environment variable that can be set to change the `xcodebuild` `-destination` parameter of one of the tests commands. This will unblock creating the bundle for SauceLabs on Buildkite, see https://github.com/wordpress-mobile/gutenberg-mobile/pull/6134

The other commits are tangential changes related to making the SauceLabs builds in CI possible or faster. Refer to the expanded commit messages for more details.

## Why?
The context of these changes is that we are incrementally moving from CircleCI to Buildkite as the CI provider for `gutenberg-mobile`. Unfortunately, there are some implicit dependencies on the CI provider and between that and this repository in the automation scripts, which is why these changes are necessary.

## How?
Moved the `test:e2e:build-app:ios` command from a long inline string in the `package.json` to a dedicated script and made it check the environment for `RN_EDITOR_E2E_IOS_DESTINATION` to set the `-destination` parameter.

## Testing Instructions




See https://github.com/wordpress-mobile/gutenberg-mobile/pull/6138 which runs the gutenberg-mobile CI against this PR.

### Testing Instructions for Keyboard

N.A.

## Screenshots or screencast 
N.A.
